app/controllers/orders_controller.rb

```diff
@@ -0,0 +1,25 @@
+# frozen_string_literal: true
+
+class OrdersController < ApplicationController
+  before_action :authenticate_employee!
+  before_action :reward_find, only: [:create]
+
+  def create
```

Comment:
FYI - there is a lot of logic in this controller, and in general this is not a good practice -> check out 'service objects' and ping me if you will have any question about ti

Reply:
Service Object implemented, you can check what you think about it :)

Reply:
Looks great :)

Reply:
btw I'm curious why you have decided to use monads :)